packages/interactivity/src/proxies/state.ts

```diff
@@ -0,0 +1,238 @@
+		proxyToProps.set( proxy, new Map() );
+	}
+	key = typeof key === 'number' ? `${ key }` : key;
+	const props = proxyToProps.get( proxy )!;
+	if ( ! props.has( key ) ) {
+		const ns = getProxyNs( proxy );
+		const prop = new PropSignal( proxy );
+		props.set( key, prop );
+		if ( initial ) {
+			const { get, value } = initial;
+			if ( get ) {
+				prop.setGetter( get );
+			} else {
+				prop.setValue(
+					shouldProxy( value ) ? proxifyState( ns, value ) : value
+				);
+			}
+		}
+	}
+	return props.get( key )!;
+};
+
+/**
+ * Relates each proxied object (i.e., the original object) with a signal that
+ * tracks changes in the number of properties.
+ */
+const objToIterable = new WeakMap< object, Signal< number > >();
+
+/**
+ * When this flag is `true`, it avoids any signal subscription, overriding state
+ * props' "reactive" behavior.
+ */
+let peeking = false;
+
+/**
+ * Handlers for reactive objects and arrays in the state.
+ */
+const stateHandlers: ProxyHandler< object > = {
+	get( target: object, key: string | symbol, receiver: object ): any {
+		/*
+		 * The property should not be reactive for the following cases:
+		 * 1. While using the `peek` function to read the property.
+		 * 2. The property exists but comes from the Object or Array prototypes.
+		 * 3. The property key is a known symbol.
+		 */
+		if (
+			peeking ||
+			( ! target.hasOwnProperty( key ) && key in target ) ||
+			( typeof key === 'symbol' && wellKnownSymbols.has( key ) )
+		) {
+			return Reflect.get( target, key, receiver );
+		}
+
+		// At this point, the property should be reactive.
+		const desc = Object.getOwnPropertyDescriptor( target, key );
+		const prop = getPropSignal( receiver, key, desc );
+
+		if ( peeking ) {
+			return prop.getComputed().peek();
+		}
+
+		const result = prop.getComputed().value;
+
+		/*
+		 * Check if the property is a synchronous function. If it is, set the
+		 * default namespace. Synchronous functions always run in the proper scope,
+		 * which is set by the Directives component.
+		 */
+		if ( typeof result === 'function' ) {
+			const ns = getProxyNs( receiver );
+			return ( ...args: unknown[] ) => {
+				setNamespace( ns );
+				try {
+					return result.call( receiver, ...args );
+				} finally {
+					resetNamespace();
+				}
+			};
+		}
+
+		return result;
+	},
+
+	set(
+		target: object,
+		key: string,
+		value: unknown,
+		receiver: object
+	): boolean {
+		setNamespace( getProxyNs( receiver ) );
+		try {
+			return Reflect.set( target, key, value, receiver );
+		} finally {
+			resetNamespace();
+		}
+	},
+
+	defineProperty(
+		target: object,
+		key: string,
+		desc: PropertyDescriptor
+	): boolean {
+		const isNew = ! ( key in target );
+		const result = Reflect.defineProperty( target, key, desc );
+
+		if ( result ) {
+			const receiver = getProxy( target );
+			const prop = getPropSignal( receiver, key );
+			const { get, value } = desc;
+			if ( get ) {
+				prop.setGetter( desc.get! );
+			} else {
```

Comment:
Does that work?

```suggestion
				prop.setGetter( get );
```

Reply:
I guess you mean the line above the `else` right? Yup, it works fine.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/62734/commits/7a3cf86076d6052b4cf416789af6f26d453cdb7f.

Reply:
Yes, sorry.